### PR TITLE
profile and property for fat-jar. Fixes #21.

### LINKED ISSUE #21: Create a dependency-bundled jar
Request to create an additional non-shaded, standalone jar for our downloads area.
* This can have namespace clashes, classpath issues, these issues exist for other projects too (such as PDFBOX using BouncyCastle, etc)
* This will result in two separate "fat" jars.  The "uber" jar, which is maven-friendly, and the "release" jar which is available on our GitHub project page for download.

---
Original bug report:

Trying to run `java-simple-serial-connector-2.8.1-SNAPSHOT.jar` inside a project causes the following exception:

```
Caused by: java.lang.NoClassDefFoundError: org/scijava/nativelib/NativeLibraryUtil
	at jssc.SerialNativeInterface.<clinit>(SerialNativeInterface.java:81)
	at jssc.SerialPortList.<clinit>(SerialPortList.java:43)
```

I consider this a bug. Previous versions of JSSC did not have this issue.

### PR DESCRIPTION
**Pull request by bmhm**
Adds a property and a profile. Both will work. E.g.

  * `mvn clean install -Ppackage,jar-with-dependencies`
  * `mvn clean install -Ppackage -Djar.dependencies.skip=false`

https://github.com/java-native/jssc/issues/21

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
     <cmake.generate.extra></cmake.generate.extra>
     <cmake.generate.skip>false</cmake.generate.skip>
     <javah.skip>false</javah.skip>
+    <skip.fat.jar>true</skip.fat.jar>
```
Comment (tresf) [inline pom.xml]: Just a thought.... `with-dependencies` may be a more intuitive name?

Comment (bmhm) [reply pom.xml]: Good idea. While "fat jar" seems to be more common in the maven commit, with-dependencies is the official term and not only sounds more proper, but also gives a hint of what is included.

Comment (bmhm) [reply pom.xml]: Done

**Comment by tresf** [comment]
I find `jar` to be redundant all profiles produce a jar.  I'd like simply `with-dependencies` if there are no objections.

**Comment by tresf** [comment]
> I find jar to be redundant all profiles produce a jar. I'd like simply with-dependencies if there are no objections.

Sorry, just realized this is a [predefined property](https://maven.apache.org/plugins/maven-assembly-plugin/descriptor-refs.html#jar-with-dependencies).  Looks good!

**Comment by tresf** [comment]
Just made a minor change to the variable name `foo.bar.skip` instead of `skip.foo.bar` to be consistent with our other variable names.

**Pull request merged**